core/store/src/db.rs

```diff
@@ -278,8 +278,8 @@ pub(crate) trait Database: Sync + Send {
         key_prefix: &'a [u8],
     ) -> Box<dyn Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a>;
     fn write(&self, batch: DBTransaction) -> Result<(), DBError>;
-    fn as_rocksdb(&self) -> Option<&RocksDB> {
-        None
+    fn flush(&self) -> Result<(), DBError> {
+        Ok(())
     }
```

Comment:
```suggestion
    fn flush(&self) -> Result<(), DBError>;
```

Mostly just pedantry, but I'd rather make the caller explicitly do nothing here, to make it impossible to forget to override the API.